### PR TITLE
httpbakery/error_test.go: Added error tests

### PR DESCRIPTION
**Pull request by mattyw**
I've added some tests for WriteDischargeRequiredError

### CODE REVIEW

File: httpbakery/error_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	m, err := macaroon.New([]byte("secret"), "id", "a location")
+	c.Assert(err, gc.IsNil)
+	tests := []struct {
+		about        string
+		path         string
+		err          error
+		expectedBody string
+	}{
+		{
+			`no path err: "an error"`,
```
Comment (alesstimec) [inline httpbakery/error_test.go]: a bit more descriptive "about", please

File: httpbakery/error_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			"",
+			errors.New("an error"),
+			fmt.Sprintf(`{"Code":"macaroon discharge required","Message":"an error","Info":{"Macaroon":{"caveats":[],"location":"%s","identifier":"%s","signature":"%v"}}}`,
+				m.Location(),
+				m.Id(),
+				hex.EncodeToString(m.Signature()),
+			),
+		},
+		{
+			`path err: "an error"`,
```
Comment (alesstimec) [inline httpbakery/error_test.go]: ditto

File: httpbakery/error_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			"http://foobar:1234",
+			errors.New("an error"),
+			fmt.Sprintf(`{"Code":"macaroon discharge required","Message":"an error","Info":{"Macaroon":{"caveats":[],"location":"%s","identifier":"%s","signature":"%v"},"MacaroonPath":"http://foobar:1234"}}`,
+				m.Location(),
+				m.Id(),
+				hex.EncodeToString(m.Signature()),
+			),
+		},
+		{
+			`path with nil err`,
```
Comment (alesstimec) [inline httpbakery/error_test.go]: ditto

File: httpbakery/error_test.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+func (s *ErrorSuite) TestWriteDischargeRequiredError(c *gc.C) {
+	m, err := macaroon.New([]byte("secret"), "id", "a location")
+	c.Assert(err, gc.IsNil)
+	tests := []struct {
+		about        string
+		path         string
+		err          error
+		expectedBody string
+	}{
+		{
```
Comment (alesstimec) [inline httpbakery/error_test.go]: reduce indent by moving { to the previous line, please

File: httpbakery/error_test.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+			`write discharge required with "an error" but no path`,
+			"",
+			errors.New("an error"),
+			fmt.Sprintf(`{"Code":"macaroon discharge required","Message":"an error","Info":{"Macaroon":{"caveats":[],"location":"%s","identifier":"%s","signature":"%v"}}}`,
```
Comment (rogpeppe) [inline httpbakery/error_test.go]: I suggest importing juju/httptesting and using AssertJSONResponse - then you can write these
as simple Error literals.

**Comment by mattyw** [comment]
@alesstimec @rogpeppe PTAL

**Comment by mattyw** [comment]
@alesstimec @rogpeppe If you're both happy with the table test approach I'll rebase down to 1 commit

**Comment by alesstimec** [comment]
:+1:

**Comment by rogpeppe** [comment]
I much prefer the table-driven test approach.
LGTM modulo a couple of minor points above.
Thanks!

**Comment by mattyw** [comment]
@rogpeppe @alesstimec thanks for the suggestions, looks better now, PTAL

**Comment by rogpeppe** [comment]
LGTM, thanks!

**Pull request merged**